core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -66,6 +72,47 @@
 
   private static final Logger log = LoggerFactory.getLogger(ApplicationServer.class);
 
+  private static final Version VERSION =
+      parseVersion(System.getProperty("java.specification.version"));
+
+  // Package private for testing
+  static Version parseVersion(String versionString) {
+    final StringTokenizer st = new StringTokenizer(versionString, ".");
+    int majorVersion = Integer.parseInt(st.nextToken());
+    int minorVersion;
+    if (st.hasMoreTokens()) {
+      minorVersion = Integer.parseInt(st.nextToken());
+    } else {
+      minorVersion = 0;
+    }
+    return new Version(majorVersion, minorVersion);
+  }
+
+  // Having these as static final provides the best opportunity for compiler optimization
+  public static final boolean IS_JAVA11_COMPATIBLE = VERSION.isJava11Compatible();
+
+  // Package private for testing
+  static class Version {
+    public final int majorVersion;
+    public final int minorVersion;
+
+    private Version(int majorVersion, int minorVersion) {
+      this.majorVersion = majorVersion;
+      this.minorVersion = minorVersion;
+    }
+
+    @Override
+    public String toString() {
+      return "Version(majorVersion=" + majorVersion
+              + ", minorVersion=" + minorVersion + ")";
+    }
+
+    // Package private for testing
+    boolean isJava11Compatible() {
+      return majorVersion >= 11;
+    }
+  }
```

Comment:
I'd prefer replacing this with a single `static` method that gets the system property, extracts the major version (post Java 8 style), and returns a `boolean`. It will be the same in terms of actual efficiency (because we need to call it just once), but it will be much simpler to read and understand.

Reply:
+1

Reply:
Done